classes/submission/PKPSubmission.inc.php

```diff
@@ -531,11 +531,11 @@ public function getAuthors($onlyIncludeInBrowse = false)
             return [];
         }
         if ($onlyIncludeInBrowse) {
-            return array_filter($authors, function ($author) {
+            return array_filter(iterator_to_array($authors), function ($author) {
                 return $author->getData('includeInBrowse');
             });
         }
-        return $authors;
+        return iterator_to_array($authors);
```

Comment:
This is another place to move away from `iterator_to_array`. Instead you can use the `->toArray()` method on the Collection:
```php
$authors = $authors->toArray();
```

Reply:
@NateWr, this function was deprecated in 3.2.0; how do you feel about just removing it outright? This feels like a time to clean out some deadweight and a couple major releases seem reasonable.

Reply:
:+1:  I'd be happy to remove it. My one hesitation is that this may be used in themes. We can remove it, but let's make sure to document that is has been removed if it effects themes.

Reply:
@defstat, If you agree, let's just remove the function entirely. @NateWr, do you have a list of notes going for the release documentation?

Reply:
Are you suggesting removing `iterator_to_array` entirely? or just from here? I will remove it, and maybe write in the issue places that may still be needed. (some of the places that I encountered the need for `iterator_to_array` was inside Grids)

Reply:
I mean the whole `getAuthors` function -- it's deprecated: https://github.com/pkp/pkp-lib/pull/7211/files/a5e7da1a1ae51b2c6f6e311cff2ac31701a6a91d#diff-054b5e15b232b71da7e2b65cd04d604d134b3df05cbc18d3eebff5efc4164806R521

Reply:
I tried to remove the references of it. I added an author\Repository method that return a collection of authors for convenience.

I did not remove the getAuthors method from the Submission class - it is being used within template files. My first thought there was to construct the author collection in the controller and pass it to the template - but this is not always possible there. I am wondering whether the `$submission->getCurrentPublication()->getData('authors')` could be a solution here.

Reply:
:+1: yes, I think that should work.

Reply:
@NateWr, @asmecher I changed it to use the `getCurrentPublication()` mostly because the original `getAuthors` was also using that - I added that in a separate commit.